tasks/create-pyxis-image/create-pyxis-image.yaml

```diff
@@ -156,19 +156,55 @@ spec:
                   ORAS_ARGS+=("--platform $os/$arch")
                 fi
 
+                # Save the OCI manifest locally, to pass to a script to create the pyxis entry
+                MANIFEST_FILE="/tmp/oras-manifest-fetch.json"
                 oras manifest fetch \
                   --registry-config "$AUTH_FILE" \
                   "${ORAS_ARGS[@]}" \
                   "${PULLSPEC}" \
-                      | tee /tmp/oras-manifest-fetch.json
+                      | tee "${MANIFEST_FILE}"
+
+                # Augment that manifest with further information about the layers, decompressed
+                # This requires pulling the layers to decompress and then measure them
+                while IFS= read -r BLOB_DETAIL;
+                do
+                    BLOB_TYPE=$(jq -r '.mediaType' <<< "$BLOB_DETAIL")
+                    BLOB_DIGEST=$(jq -r '.digest' <<< "$BLOB_DETAIL")
+
+                    if [[ "$BLOB_TYPE" =~ ^.*\.gzip$|^.*\+gzip$ ]]; then
+                        BLOB_FILE="/tmp/oras-blob-fetch-${BLOB_DIGEST}"
+                        BLOB_PULLSPEC="${PULLSPEC%%@*}@${BLOB_DIGEST}"
+
+                        # Save the blob
+                        oras blob fetch \
+                          --registry-config "$AUTH_FILE" \
+                          --output "${BLOB_FILE}.gz" \
+                          "${BLOB_PULLSPEC}"
+
+                        # Decompress it
+                        gunzip "${BLOB_FILE}.gz"
+
+                        # Measure it
+                        EXPANDED_DIGEST=$(sha256sum "${BLOB_FILE}" | cut -d " " -f 1)
+                        EXPANDED_SIZE=$(wc --bytes "${BLOB_FILE}" | awk '{print $1}' | tr -d '\n')
+
+                        # Append this information to the parsed_data manifest
+                        cat <<< $(jq \
+                          '.uncompressed_layers += [{"digest": "'"$EXPANDED_DIGEST"'", "size": '"$EXPANDED_SIZE"'}]' \
```

Comment:
So if they are not compressed, the thing that `oras manifest fetch` returns will already include `uncompressed_layers`?

Reply:
Hm, no - if the layers are not compressed, then `oras manifest fetch` will return a `.layers` array, and it will include uncompressed sizes for each layer, but it wouldn't store them in a `.uncompressed_layers` key.

This whole "uncompressed layers" is a RH thing as far as I can tell, used only in pyxis and the RH catalog. I haven't found any precedent in the OCI world.

In any case, if the layers happen to be already uncompressed (which is possible) or if they are compressed with some compression other than gzip, then this whole codepath should be skipped and we will *omit* `uncompressed_layers` from the data handed to `create_pyxis_image.py`. It is likely that this will happen for all sorts non-image OCI artifacts (like models and disk images) and in the future we might switch compression schemes like in https://github.com/konflux-ci/build-definitions/issues/1264, so we should be careful in https://github.com/konflux-ci/release-service-utils/pull/284 to make sure that `create_pyxis_image.py` can proceed happily when `uncompressed_layers` is absent and be careful here to make sure that the task can complete without processing any `uncompressed_layers`.

Reply:
> if the layers happen to be already uncompressed (which is possible) or if they are compressed with some compression other than gzip, then this whole codepath should be skipped

OK, the way I read it was that the "if compressed" is there to cover the use case when they are compressed. But if uncompressed, it would just work. So now I see that't the case. (And I guess in reality, normal images have the layers always gzipped.)

So maybe make it clearer in the commit/pr description? Like add "if uncompressed, the uncompressed data will not be saved" or something. And maybe also add a comment in the code about this somewhere above the `if`? Like "normal images will always have the layers compressed. if they are not compressed, don't save the uncompressed data in Pyxis". Something like that.

Reply:
Done in 4e015ed.